app/controllers/structures_controller.rb

```diff
@@ -32,12 +32,20 @@ def update
   end
 
   def hierarchy
-    @cocina_item = Repository.find(decrypted_token.fetch(:druid))
+    @cocina_item = fetch_cocina(item_id:, user_version_id:)
     @root_directory = FileHierarchyService.to_hierarchy(cocina_object: @cocina_item)
   end
 
   private
 
+  def item_id
+    decrypted_token&.fetch(:druid, nil) || params[:item_id]
+  end
+
+  def user_version_id
+    decrypted_token&.fetch(:user_version_id, nil) || params[:user_version_id]
+  end
```

Comment:
If we combine these two modes (get a value from a token, else get it from params), is there a risk of going around the token-based method? Presumably we had a good reason for getting values from encrypted tokens. If we're already doing this elsewhere and we dismissed this as a risk already, feel free to disregard!

Reply:
Good question, I could revert the change at line 12 that uses param for the csv, but I wonder if it should use the token and then remove the use of params.

Reply:
I don't have a good handle on why/how we're using the token here, so maybe?

Reply:
Yeah, I agree with Mike. I think the DRYing allows bypassing the tokens. I think it would be easier to understand if getting cocina from params and tokens were separate (or if `find_cocina` took druid and version parameters).